src/packaging/kpatch/commit_to_overlay.sh

```diff
@@ -0,0 +1,459 @@
+    done
+
+    echo -en "\n"
+    read -p "Is this correct? [y - proceed | n - exit]: " user_input
+    case "${user_input}" in
+      "y"*|"Y"*)
+        echo "Proceeding."
+        ;;
+      "n"*|"N"*)
+        echo "Exiting."
+        exit
+        ;;
+      *)
+        echo -en "${COLOR_RED}ERROR: Invalid response.\nPlease enter 'y' to proceed, or 'n' to" \
+                 "exit.${NO_COLOR}\n\n"
+        ;;
+    esac
+  done
+}
+
+# Prompt the user to push the overlay branch upstream
+prompt_user_push() {
+  user_input=""
+
+  while [[ ! ${user_input} =~ ^(y|Y|m|M) ]]; do
+    echo -en "\nPlease verify everything expected has been applied to overlay branch" \
+             "${OVERLAY_BRANCH}.\n"
+    echo "Ready to push ${OVERLAY_BRANCH} to $(get_repo_name ${KERNEL_REPO_URL})?"
+    echo -en "[y - yes, push | n - no, delete | m - keep branch for manual push later]: "
+    read user_input
+
+    case "${user_input}" in
+      "y"*|"Y"*)
+        echo -en "Proceeding with push.\n\n"
+        ;;
+      "n"*|"N"*)
+        exit
+        ;;
+      "m"*|"M"*)
+        echo "Retaining ${LINUX_SRC} and branch ${OVERLAY_BRANCH} for manual push later."
+        MANUAL_PUSH=0
+        ;;
+      *)
+        echo -en "${COLOR_RED}ERROR: Invalid response.\nPlease enter 'y' to proceed with push," \
+                 "'n' to delete the branch and exit,\nor 'm' to retain the branch for manual" \
+                 "push at a later time.${NO_COLOR}\n\n"
+        ;;
+    esac
+  done
+}
+
+# Cleanup artifacts
+_cleanup() {
+  cd $RUN_DIR
+  echo -en "\nCleaning up and exiting\n"
+
+  if [[ -d ${LINUX_SRC} ]] && [[ -d ${VDO_TREE} ]] && [[ ${DEBUG} != 0 ]]; then
+    if [[ ${MANUAL_PUSH} != 0 ]]; then
+      echo "Removing ${LINUX_SRC}"
+      rm -rf ${LINUX_SRC}
+    fi
+
+    echo "Removing ${VDO_TREE}"
+    rm -rf ${VDO_TREE}
+  fi
+
+  if [[ -e ${commit_file} ]] && [[ -f ${commit_file} ]]; then
+    rm -fv ${commit_file}
+  fi
+
+  unset -f CHANGE_LOG DEBUG GITDATE GITAUTHOR LINUX_SRC
+}
+
+###############################################################################
+# main()
+trap _cleanup EXIT
+test "$#" -lt 5 && print_usage
+
+process_args $@
+print_config
+checkout_source_branch
+$PROCESS_INPUT_FX
+
+echo # Intentional blank line
+
+# Store linux repo location as an absolute path in the current directory
+LINUX_SRC=${RUN_DIR}/$(mktemp -d vdo-kernel-XXXXXX)
+VDO_TREE=${RUN_DIR}/$(mktemp -d vdo-raw-XXXXXX)
+
+# Shallow clone the specified dm-linux repo branch into LINUX_SRC
+echo "Cloning kernel repo $(get_repo_name ${KERNEL_REPO_URL})/${KERNEL_BRANCH}"
+git clone --branch ${KERNEL_BRANCH} --depth 1 ${KERNEL_REPO_URL} ${LINUX_SRC}
+cd $LINUX_SRC
+git checkout -b ${OVERLAY_BRANCH} ${KERNEL_BRANCH}
+
+# Need to detect that our repo is x number of commits ahead of dm-vdo/vdo-devel
+# Clone the current repo as ${VDO_TREE}? Is VDO_TREE even needed? Aren't we working from the clone when this script is called? Need to make a copy?
```

Comment:
I agree we probably don't need this clone but we can figure that out later.

Reply:
True. Probably good to have the option to clone or enter a path to an existing clone, as we had discussed earlier. Maybe an option isn't needed, but that depends on the workflow we decide to follow.